src/actions/arcActions.ts

```diff
@@ -632,37 +629,32 @@ export function createDAO(daoName: string, tokenName: string, tokenSymbol: strin
 
 export type CreateProposalAction = IAsyncAction<'ARC_CREATE_PROPOSAL', { avatarAddress: string }, any>;
 
-export function createProposal(daoAvatarAddress: string, title: string, description: string, nativeTokenReward: number, reputationReward: number, ethReward: number, externalTokenReward: number, beneficiaryAddress: string): ThunkAction<any, IRootState, null> {
+
+export function createProposal(daoAvatarAddress: string, title: string, url: string, nativeTokenReward: number, reputationReward: number, ethReward: number, externalTokenReward: number, beneficiaryAddress: string): ThunkAction<any, IRootState, null> {
   return async (dispatch: Redux.Dispatch<any>, getState: () => IRootState) => {
     const meta = {
       avatarAddress: daoAvatarAddress
     };
 
     try {
       const web3: Web3 = await Arc.Utils.getWeb3();
-
-      if (!beneficiaryAddress.startsWith("0x")) { beneficiaryAddress = "0x" + beneficiaryAddress; }
-      beneficiaryAddress = beneficiaryAddress.toLowerCase();
-
-      const ethAccountAddress: string = getState().web3.ethAccountAddress;
-      const dao = getState().arc.daos[daoAvatarAddress];
-
-      const contributionRewardInstance = await Arc.ContributionRewardFactory.deployed();
-
-      const descriptionHash = Arc.Utils.SHA3(description);
       const submittedTime = Math.round((new Date()).getTime() / 1000);
 
-      // Save the proposal title, description and submitted time on the server
+      const description = {
+        title,
+        url,
+        daoAvatarAddress,
```

Comment:
why do we need this?

Reply:
Following the convention set by the alchemy-server which stores: arcId, daoAvatarAddress, descriptionHash, description, submittedAt, and an id. To help this new data store fit into the existing system, I added the same fields with slight modifications eg renaming description to url.